backend/src/routes/yelpFusion.ts

```
@@ -5,12 +5,22 @@ const router = express.Router();
 const yelp = require("yelp-fusion");
 const apiKey = process.env.YELP_API_KEY;
 const client = yelp.client(apiKey);
+
 const updateItinerary = async (
+  req: any,
   itineraryId: string,
   activityId: string,
   businessIds: string[]
 ) => {
-  const itinerary = await Itinerary.findOne({ _id: itineraryId });
+  const itinerary = await Itinerary.findOne({
+    $or: [
+      { _id: itineraryId, user_id: req.session.userId },
+      {
+        _id: itineraryId,
+        collaborators: { $elemMatch: { _id: req.session.userId } },
+      },
+    ],
+  });
```

Since the APIs like /restaruants is surrounded in try catch, lets throw an error here if there is no existing itinerary (and it will be caught in the API endpoint)

something like if (!itinerary) throw new Error("No itinerary found")

make sure to test when it's thrown though, it should be caught in our API